swatch-picker/open-swatch-picker.js

```diff
@@ -0,0 +1,202 @@
+import { oklchToHex, oklchToRgb } from "./color-converters.js";
+
+
+const styles = new CSSStyleSheet()
+styles.replaceSync(/* css */`
+    dialog {
+        border-radius: 1rem;
+
+        &::backdrop {
+            background: rgba(0, 0, 0, 0.5);
+            opacity: 0.75;
+        }
+
+        & div[scale] {
+            display: flex;
+
+            & h2 {
+                display: none;
+            }
+            & h2[active] {
+                grid-column: span 13;
+                display: block;
+            }
+            & div[swatch] {
+                margin: 2px;
+
+                & button {
+                    width: 2rem;
+                    height: 2rem;
+                    border: none;
+                    border-radius: 0.5rem;
+                    cursor: pointer;
+                    outline: 2px solid transparent;
+
+                    &:hover, &:focus {
+                        outline: 2px solid;
+                    }
+                }
+                & span {
+                    display: none;
+                }
+            }
+        }
+    }
+    [part="swatch-picker"] {
+        display: flex;
+        align-items: center;
+        gap: 1rem;
+    }
+`);
+const scales = ['neutral', 'stone', 'slate', 'red', 'orange', 'amber', 'gold', 'yellow', 'lime', 'green', 'emerald', 'teal', 'cyan', 'sky', 'blue', 'violet', 'purple', 'fuchsia', 'magenta', 'pink', 'rose'];
+    
+const makeScale = (name) => {
+    let length = 12;
+    if( name === 'neutral' ) length = 13;
+    return /* html */`
+    <div scale>
+        <h2>${name}</h2>
+        ${Array.from({ length: length }, (_, i) => makeColorSwatch(name, i + 1)).join('')}
+    </div>`
+}
+
+const makeColorSwatch = (name, i) => /* html */`
+    <div swatch>
+        <button aria-label="Copy ${name} 1" style="background:var(--${name}-${i})">
+            &nbsp;
+        </button>
```

Comment:
Nit, nbsp not necessary:

```suggestion
        <button aria-label="Copy ${name} 1" style="background:var(--${name}-${i})"></button>
```